get_http_test.go

```diff
@@ -404,6 +406,35 @@ func TestHttpGetter_cleanhttp(t *testing.T) {
 	}
 }
 
+func TestHttpGetter__RespectsContextCanceled(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel() // cancel immediately
+
+	ln := testHttpServer(t)
+
+	var u url.URL
+	u.Scheme = "http"
+	u.Host = ln.Addr().String()
+	u.Path = "/sleep"
+	dst := tempDir(t)
+
+	g := new(HttpGetter)
+	g.client = &Client{
+		Ctx: ctx,
+	}
+
+	start := time.Now()
+	err := g.Get(dst, &u)
+	dur := time.Since(start)
+
+	if !errors.Is(err, context.Canceled) {
+		t.Fatalf("expected context.Canceled, got: %v", err)
+	}
+	if dur > 1*time.Second {
+		t.Fatal("Get() took too long")
```

Comment:
This test might get flaky with CI tools and I think that the http client will return before actually sending the request, so the err check should be enough IMO.
If you agree then let's remove it ?

Reply:
At least in my production code I got the context.Canceled due to the copy operations later aborting and returning the context.Canceled. 
The important part here is that we use the `ctx` in the `http.Request`. We could write our own mock http.RoundTripper and test the context there, but I am not sure if that isn't too focused.

Reply:
I was trying to say that time-bound testing is not always a good way to test because often: ci tools are shared instances, and sometimes they can just randomly freeze for a (possibly long) moment.

Reply:
> and sometimes they can just randomly freeze for a (possibly long) moment.

Agreed. I will rework the code a bit to explicitly check in the transport or sthg. Stay tuned 👍

Reply:
But the above tests line #430 sorts of guarantees that this is valid, because the error has to be a `context.Canceled` one.